test/src/test/java/hudson/model/RSSTest.java

```diff
@@ -135,6 +201,86 @@ public void latestBuilds() throws Exception {
         assertEquals(6, allLinks.getLength());
     }
 
+    private void checkRssBasicNodes(Node channelNode, String expectedTitle, int expectedNodes) throws IOException {
+        assertThat(channelNode.getChildNodes().getLength(), is(expectedNodes));
+        assertThat(getSingleNode(channelNode, "link").getTextContent(), is(j.getURL().toString()));
+        assertThat(getSingleNode(channelNode, "description").getTextContent(), is(expectedTitle));
+        assertThat(getSingleNode(channelNode, "title").getTextContent(), is(expectedTitle));
+    }
+
+    private void checkAtomBasicNodes(Node feedNode, String expectedTitle, int expectedNodes) throws IOException {
+        assertThat(feedNode.getChildNodes().getLength(), is(expectedNodes));
+        Node linkNode = getSingleNode(feedNode, "link");
+        assertThat(linkNode.getAttributes().getNamedItem("rel").getTextContent(), is("alternate"));
+        assertThat(linkNode.getAttributes().getNamedItem("type").getTextContent(), is("text/html"));
+        assertThat(linkNode.getAttributes().getNamedItem("href").getTextContent(), is(j.getURL().toString()));
+        assertNotNull(getSingleNode(feedNode, "updated"));
+        assertThat(getSingleNode(feedNode, "title").getTextContent(), is(expectedTitle));
+        Node authorNode = getSingleNode(feedNode, "author");
+        NodeList authorNodes = authorNode.getChildNodes();
+        assertThat(authorNodes.getLength(), is(1));
+        Node nameNode = authorNodes.item(0);
+        assertThat(nameNode.getTextContent(), is("Jenkins Server"));
+        Node idNode = getSingleNode(feedNode, "id");
+        assertFalse(idNode.getTextContent().isEmpty());
+    }
+
+    private Node getSingleNode(Node parentNode, String nodeName) {
+        Node childNode = null;
+        NodeList childNodes = parentNode.getChildNodes();
+        for (int i = 0; i < childNodes.getLength(); i++) {
+            if (childNodes.item(i).getNodeName().equals(nodeName)) {
+                if (childNode == null) {
+                    childNode = childNodes.item(i);
+                } else {
+                    fail("Too many children.");
+                }
+            }
+        }
+        return childNode;
+    }
+
+    private void checkRssTimeNode(Node firstBuild, String nodeName) throws ParseException {
+        String pubDate = getSingleNode(firstBuild, nodeName).getTextContent();
+        assertNotNull(pubDate);
+        DateFormat formatter = new SimpleDateFormat("EEE, dd MMM yyyy HH:mm:ss zzz");
+        Date date = formatter.parse("Sat, 24 Apr 2010 14:01:00 GMT");
```

Comment:
```suggestion
        Date date = formatter.parse(pubDate);
```

Otherwise the test does not test what you want.

Additionnally, the `parse` will throw an exception if the date is not parseable, so the `assertNotNull` does not provide any value except confusion.

Reply:
I tried with and without. It felt kind of weird and more confusing to have it there without any assertion.